### PR TITLE
ddl: forbid disabling concurrent DDL if MDL is enabled

### LINKED ISSUE #39131: ddl blocked when the tidb_enable_concurrent_ddl is set to off
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

```
set global tidb_enable_concurrent_ddl=off;
create database if not exists test2;
```

### 3. What did you see instead (Required)
```
MySQL [test]> show processlist;
+---------------+------+------------------+------+---------+------+------------+-------------------------------------+
| Id            | User | Host             | db   | Command | Time | State      | Info                                |
+---------------+------+------------------+------+---------+------+------------+-------------------------------------+
| 2199023255955 | root | 172.17.0.1:38336 | test | Query   |  228 | autocommit | create database if not exists test2 |
| 2199023255957 | root | 172.17.0.1:38534 | test | Query   |    0 | autocommit | show processlist                    |
+---------------+------+------------------+------+---------+------+------------+-------------------------------------+
2 rows in set (0.01 sec)
```
### 4. What is your TiDB version? (Required)

```
Release Version: v6.5.0-alpha-55-gb0c338139
Edition: Community
Git Commit Hash: b0c338139c344df751e6738dea9d6fb983e108a0
Git Branch: master
UTC Build Time: 2022-11-14 07:08:30
GoVersion: go1.19.1
Race Enabled: false
TiKV Min Version: 6.2.0-alpha
Check Table Before Drop: false
Store: unistore
```

### PR DESCRIPTION
**Pull request by wjhuang2016**
Signed-off-by: wjhuang2016 <huangwenjun1997@gmail.com>

### What problem does this PR solve?

Issue Number: close #39131

forbid disabling concurrent DDL if MDL is enabled

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: ddl/concurrentddltest/switch_test.go [hunk 1]
```diff
@@ -134,3 +135,15 @@ func TestConcurrentDDLSwitch(t *testing.T) {
...
 	}
 }
+
+func TestConcurrentDDLSwitchWithMDL(t *testing.T) {
+	if !variable.EnableConcurrentDDL.Load() {
+		t.Skip("skip test if concurrent DDL is disabled")
+	}
+	store := testkit.CreateMockStore(t)
+	tk := testkit.NewTestKit(t, store)
+	tk.MustGetErrMsg("set global tidb_enable_concurrent_ddl=off", "can not disable concurrent ddl when metadata lock is enabled")
```
Comment (Benjamin2037) [inline ddl/concurrentddltest/switch_test.go:145]:
```suggestion
	tk.MustGetErrMsg("set global tidb_enable_concurrent_ddl=off", "Can not disable concurrent ddl when metadata lock is enabled")
```

File: ddl/ddl.go [hunk 2]
```diff
@@ -1198,6 +1198,10 @@ func (d *ddl) SwitchConcurrentDDL(toConcurrentDDL bool) error {
 		})
 	}
 
+	if variable.EnableMDL.Load() && !toConcurrentDDL {
+		return errors.New("can not disable concurrent ddl when metadata lock is enabled")
```
Comment (Benjamin2037) [inline ddl/ddl.go:1202]: Do we have a place to collect all error msg?

[the remark this answers is not in the document]
Comment (wjhuang2016) [reply ddl/ddl.go:1202]: We prefer the lower case pattern for the error message.

Comment (wjhuang2016) [reply ddl/ddl.go:1202]: For standard error with the error code, yes. For these misc errors, no.

**Review by tangenta — APPROVED** [review]

**Pull request merged**